File: src/main/java/ink/rubi/coffee/po/holder/TemplateConfigHolder.java
package ink.rubi.coffee.po.holder;

import com.baomidou.mybatisplus.generator.config.TemplateConfig;
import lombok.Data;
import lombok.experimental.Accessors;

/**
 * @author Rubi
 * @since 2019-06-05 15:43
 */
@Data
@Accessors(chain = true)
public class TemplateConfigHolder implements ConfigHolder<TemplateConfig> {

    private String entity;

    private String entityKt;

    private String service;

    private String serviceImpl;

    private String mapper;

    private String xml;

    private String controller;

    private String engine;


    @Override
    public TemplateConfig convert() {
        return new TemplateConfig()
                .setEntity(this.entity)
                .setEntityKt(this.entityKt)
                .setServiceImpl(this.serviceImpl)
                .setService(this.service)
                .setMapper(this.mapper)
                .setXml(this.xml)
                .setController(this.controller);
    }


}

File: src/main/java/ink/rubi/coffee/po/holder/TemplateConfigHolder.kt
package ink.rubi.coffee.po.holder

import com.baomidou.mybatisplus.generator.config.TemplateConfig

/**
 * @author Rubi
 * @since 2019-06-05 15:43
 */

data class TemplateConfigHolder(
        var entity: String? = null,
        var entityKt: String? = null,
        var service: String? = null,
        var serviceImpl: String? = null,
        var mapper: String? = null,
        var xml: String? = null,
        var controller: String? = null,
        var engine: String? = null) : ConfigHolder<TemplateConfig> {


    override fun convert(): TemplateConfig {
        return TemplateConfig()
                .setEntity(this.entity)
                .setEntityKt(this.entityKt)
                .setServiceImpl(this.serviceImpl)
                .setService(this.service)
                .setMapper(this.mapper)
                .setXml(this.xml)
                .setController(this.controller)
    }


}
